libbeat/cmd/instance/beat.go

```diff
@@ -879,6 +878,24 @@ func (b *Beat) configure(settings Settings) error {
 	return err
 }
 
+func (b *Beat) initConfigManager(forcePackageVersion bool) error {
+	// initialize config manager
+	m, err := management.NewManager(b.Config.Management, reload.RegisterV2)
+	if err != nil {
+		return err
+	}
+	b.Manager = m
+
+	// The manager initialisation reads the package version sent by the agent,
+	// now the manager is initialised, we apply the package version if the agent
+	// sent it.
```

Comment:
```suggestion
	// The manager initialisation reads the package version sent by the agent,
	// now the manager is initialised, we apply the package version if the agent
	// sent it.
	//
	// Necessary because the Beat version may not match the agent version, but from
	// the user's perspective there is only one version in the system - the Elastic
	// version of the Agent package (.tar.gz, .zip, etc.) that they installed.
	// Make sure that is the version that is presented to the user in published events.
```

Let's add some more explanation about why we do this.